src/form-engine.component.tsx

```diff
@@ -142,6 +150,18 @@ const FormEngine: React.FC<FormProps> = ({
     reportError(patientError, t);
   }, [patientError, t]);
 
+  useEffect(() => {
+    if (!isLoadingFormJson) {
+      refinedFormJson?.pages.forEach((page) => {
+        page.sections.forEach((section) => {
+          if (section.questions.some((question) => question.type === 'programWorkflow')) {
+            setHasPatientWorkflows(true);
+          }
+        });
+      });
+    }
+  }, [isLoadingFormJson]);
```

Comment:
I would let the `usePatientEnrollment(..)` hook handle this. The idea is to conditionally load data.